packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -396,24 +405,64 @@ const SynapseGrid = forwardRef<
                 value={rowValues}
                 columns={colValues}
                 rowKey="_rowId"
-                rowClassName={({ rowData }: any) => {
+                rowClassName={({ rowData, rowIndex }: any) => {
+                  let className = ''
                   if (deletedRowIds.has(rowData._rowId)) {
-                    return 'row-deleted'
+                    className += ' row-deleted'
                   }
                   if (createdRowIds.has(rowData._rowId)) {
-                    return 'row-created'
+                    className += ' row-created'
                   }
                   if (updatedRowIds.has(rowData._rowId)) {
-                    return 'row-updated'
-                  } else return ''
+                    className += ' row-updated'
+                  }
+                  if (rowData.validationStatus === 'valid')
+                    className += ' row-valid'
+                  if (rowData.validationStatus === 'invalid')
+                    className += ' row-invalid'
+                  if (rowData.validationStatus === 'unknown')
+                    className += ' row-unknown'
+                  if (selectedRowIndex === rowIndex) {
+                    className += '  row-selected'
+                  }
+                  return className
                 }}
                 createRow={addRowToModel}
                 duplicateRow={({ rowData }: any) => ({
                   ...rowData,
                   _rowId: genId(),
                 })}
                 onChange={handleChange}
+                onActiveCellChange={({ cell }) => {
+                  setSelectedRowIndex(cell ? cell.row : null)
+                }}
               />
+              {/* Show validation messages for selected row */}
+              {selectedRowIndex !== null &&
+                rowValues[selectedRowIndex] &&
+                Array.isArray(rowValues[selectedRowIndex].validationMessages) &&
+                rowValues[selectedRowIndex].validationMessages.length > 0 && (
+                  <div
+                    style={{
+                      marginTop: 12,
+                      background: '#fffbe6',
+                      border: '1px solid #ffe58f',
+                      borderRadius: 4,
+                      padding: 10,
+                      maxWidth: 400,
+                      fontSize: 13,
+                    }}
+                  >
+                    <strong>Validation Messages:</strong>
+                    <ul style={{ margin: 0, paddingLeft: 18 }}>
+                      {rowValues[selectedRowIndex].validationMessages.map(
+                        (msg: string, i: number) => (
+                          <li key={i}>{msg}</li>
+                        ),
+                      )}
+                    </ul>
+                  </div>
+                )}
             </div>
```

Comment:
You could use our `FullWidthAlert` to get styling that will feel more 'native' to Synapse (and let you remove a lot of these inline styles)